src/contract.ts

```diff
@@ -119,31 +123,57 @@ export class Contract {
   }
 
   private parseResponseField(
-    element: AbiEntry | FunctionAbi,
-    responseIterator: Iterator<string>
-  ): Args {
-    let entries: AbiEntry[] = [];
-    if (['felt', 'felt*'].includes(element.type)) {
-      return responseIterator.next().value;
+    responseIterator: Iterator<string>,
+    output: AbiEntry,
+    parsedResult?: Args
+  ): any {
+    const { name, type } = output;
+    let arrLen: number;
+    const parsedDataArr: (BigNumberish | Struct)[] = [];
+    switch (true) {
+      case /_len/.test(name):
+        return parseFelt(responseIterator.next().value).toNumber();
+      case /\(felt/.test(type):
+        return type.split(',').reduce((acc) => {
+          acc.push(parseFelt(responseIterator.next().value));
+          return acc;
+        }, [] as BigNumberish[]);
+      case /\*/.test(type):
+        if (parsedResult && parsedResult[`${name}_len`]) {
+          arrLen = parsedResult[`${name}_len`] as number;
+          for (let i = 0; i < arrLen; i += 1) {
```

Comment:
I think rather than using `for loop`, we should use `while` in the following way

```
while(parsedDataArr.len < arrLen) {

parsedDataArr.push(this.parseStructOrFelt(responseIterator, output))

}
```